foreman/client.py

```diff
@@ -222,7 +247,8 @@ def __init__(self, foreman):
         """
         self._f = foreman
         # Preserve backward compatibility with old interface
-        for method_name in ('index', 'show', 'update', 'destroy', 'create'):
+        for method_name in ('index', 'show', 'update', 'destroy', 'create',
+                            'search'):
```

Comment:
I have to agree, I wasn't familiar with your example and i think it might be the same as mine.
but,
Please try to run:
>>> client.operatingsystems.index(search='')
...
TypeError: index() takes at least 2 arguments (2 given)

Why? because for some reason operatingsystems.index as well as environments.index (and mybe more objects) required a mandatory param.
This is a bug or not, but my code actually solve this. by:
operatingsystems.search('')